src/main/java/com/mongodb/quickstart/Update.java

```diff
@@ -0,0 +1,76 @@
+package com.mongodb.quickstart;
+
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.model.FindOneAndUpdateOptions;
+import com.mongodb.client.model.ReturnDocument;
+import com.mongodb.client.model.UpdateOptions;
+import com.mongodb.client.result.UpdateResult;
+import org.bson.Document;
+import org.bson.conversions.Bson;
+import org.bson.json.JsonWriterSettings;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static com.mongodb.client.model.Filters.and;
+import static com.mongodb.client.model.Filters.eq;
+import static com.mongodb.client.model.Updates.*;
+
+public class Update {
+
+    public static void main(String[] args) {
+        Logger.getLogger("org.mongodb.driver").setLevel(Level.ALL);
+        JsonWriterSettings prettyPrint = JsonWriterSettings.builder().indent(true).build();
+
+        try (MongoClient mongoClient = MongoClients.create(System.getProperty("mongodb.uri"))) {
+            MongoDatabase sampleTrainingDB = mongoClient.getDatabase("sample_training");
+            MongoCollection<Document> gradesCollection = sampleTrainingDB.getCollection("grades");
+
+            // update one document
+            Bson filter = eq("student_id", 10000);
+            Bson updateOperation = set("comment", "You should learn MongoDB!");
+            UpdateResult updateResult = gradesCollection.updateOne(filter, updateOperation);
+            System.out.println("=> Updating the doc with {\"student_id\":10000}. Adding comment.");
+            System.out.println(gradesCollection.find(filter).first().toJson(prettyPrint));
+            System.out.println(updateResult);
+
+            // upsert
+            filter = and(eq("student_id", 10002d), eq("class_id", 10d));
+            updateOperation = push("comments", "You will learn a lot if you read the MongoDB blog!");
+            UpdateOptions options = new UpdateOptions().upsert(true);
+            updateResult = gradesCollection.updateOne(filter, updateOperation, options);
+            System.out.println("\n=> Upsert document with {\"student_id\":10002.0, \"class_id\": 10.0} because it doesn't exist yet.");
```

Comment:
Why doubles not ints?

Reply:
Maybe I should use doubles everywhere then to avoid the distraction with the type equivalence when doing the queries?

Reply:
I'm happy either way - just wanted to flag it.